lib/src/shortest_path.dart

```diff
@@ -24,12 +24,12 @@ import 'dart:collection';
 ///
 /// If you supply one of [equals] or [hashCode], you should generally also to
 /// supply the other.
-List<T> shortestPath<T>(
+List<T>? shortestPath<T>(
   T start,
   T target,
   Iterable<T> Function(T) edges, {
-  bool Function(T, T) equals,
-  int Function(T) hashCode,
+  bool Function(T?, T?)? equals,
```

Comment:
I wouldn't expect the `T?` here, just `T`. Do you know why it needs to be `T?`, or what breaks if it is just `T`?

Reply:
just T